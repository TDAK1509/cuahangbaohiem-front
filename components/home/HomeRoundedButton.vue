<template>
  <div class="home-rounded-button">{{ label }}</div>
</template>

<script>
export default {
  name: "HomeRoundedButton",

  props: {
    label: {
      type: String,
      default: ""
    }
  }
};
</script>

<style lang="scss" scoped>
@import "@/assets/scss/_colors";

.home-rounded-button {
  border-radius: 50%;
  border: 2px solid $grey;
  width: 100px;
  height: 100px;
  display: flex;
  align-items: center;
  justify-content: center;
}
</style>
