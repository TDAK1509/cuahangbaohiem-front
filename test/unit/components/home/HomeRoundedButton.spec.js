import { mount } from "@vue/test-utils";
import HomeRoundedButton from "@/components/home/HomeRoundedButton.vue";

describe("HomeRoundedButton.vue", () => {
  it("matches snapshot", () => {
    const wrapper = mount(HomeRoundedButton, {
      propsData: {
        to: "/to-link",
        label: "Label",
        iconClass: "icon-class",
        iconColorClass: "icon-color-class"
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
