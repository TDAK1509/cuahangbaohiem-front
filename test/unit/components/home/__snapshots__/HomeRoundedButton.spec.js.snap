// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeRoundedButton.vue matches snapshot 1`] = `
<nuxt-link
  class="home-rounded-button"
  to="/to-link"
>
  <div>
    <span
      class="icon is-large icon-color-class"
    >
      <i
        class="fa-3x icon-class"
      />
    </span>
  </div>
   
  <p
    class="is-size-7"
  >
    Label
  </p>
</nuxt-link>
`;
